packages/commonwealth/client/scripts/models/IWebWallet.ts

```diff
@@ -9,13 +10,13 @@ interface IWebWallet<AccountT extends { address: string } | string> {
   enabling: boolean;
   accounts: readonly AccountT[];
   enable: () => Promise<void>;
-  signWithAccount: (account: Account) => Promise<string>;
   validateWithAccount: (
     account: Account,
     walletSignature: string
-  ) => Promise<void>;
-  reset?: () => Promise<void>;
-
+  ) => Promise<void>; // TODO add optional parameter: Function Callback
+  contractCall?: (tx: TransactionConfig) => Promise<string>;
```

Comment:
We shouldn't add these as global types for the IWebWallet unless we're planning on implementing them across all wallets. Instead, we should expose an API getter, which gets the wallet's API connection rather than the one we have via ChainNode.